Provide PHP 8.1 support

weierophinney opened the pull request:
This patch provides PHP 8.1 support, by making  the following changes:

- Adds 8.1 to the PHP constraint
- Changes how we replace zend-escaper
  - Renames "replace" to "conflict" in `composer.json`, and sets the constraint for "zend-escaper" to "*"
  - Removes laminas-zendframework-bridge requirement
- Adds Laminas CI configuration to ignore PHP platform req on PHP 8.1 builds

Fixes #13

weierophinney committed:
feat: prepare for PHP 8.1 support

- Adds 8.1 to the PHP constraint
- Changes how we replace zend-escaper
  - Renames "replace" to "conflict" in `composer.json`, and sets the constraint for "zend-escaper" to "*"
  - Removes laminas-zendframework-bridge requirement
- Adds Laminas CI configuration to ignore PHP platform req on PHP 8.1 builds

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
This patch also resolves #13

Ocramius commented:
Psalm seems to complain in this one:

```

Error: Syntax error, unexpected '=' on line 125
```

Ocramius reviewed (changes requested):
Psalm seems to complain in this one:

```

Error: Syntax error, unexpected '=' on line 125
```

weierophinney committed:
qa: update composer.lock

In part to ensure that psalm is running with the same version on the CI platform.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: add PHP 8 compat error to psalm baseline

Syntax flagged is valid under PHP 8, which is the only time the symfony/polyfill-mbstring bootstrap80.php file should be loaded.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
> Psalm seems to complain in this one:

Yep - it's a PHP 8-specific symfony/polyfill-mbstring file; it _should_ be excluded from our scans (as it's in the vendor tree), but for some reason is still showing. I updated the baseline so that it gets ignored, and tests are now green.

Ocramius commented on psalm-baseline.xml:
```diff
@@ -51,4 +51,9 @@
       <code>testUnicodeCodepointConversionToUtf8</code>
     </MissingReturnType>
   </file>
+  <file src="vendor/symfony/polyfill-mbstring/bootstrap80.php">
```
O_O what the heck is this doing here?
